docker/pypi/wmagent/run.sh

```diff
@@ -1,5 +1,29 @@
 #!/bin/bash
 
+_find_child_pss() {
+    # An auxiliary function to find all processes forked from the shell parrent
+    # of the current process
+    cat  /proc/$PPID/task/*/children
+}
+
+_service_gracefull_exit() {
+    # An auxiliary function to handle the WMAgent service graceful exit
+    manage stop-agent
+    wait
+
+    for proc in _find_child_pss
+    do
+        kill -9 $proc
+    done
+}
+
+# Here to define the signal we are about to trap for handling WMAgent graceful exit
+# NOTE: curently (just for testing purposes) we stick to `SIGUSR1` later we should
+#       move to SIGTERM and/or SIGKILL. In order to test the current implementation execute:
+#       docker exec -it wmagent bash
+#       kill -s 10 1
+trap _service_gracefull_exit SIGUSR1
```

Comment:
Is it the signal generated by "docker stop"?

Reply:
any `SIGUSR1` signal that  comes to the process would be trapped. I think the signal itself should be propagated  from the docker engine to the entry point, yes.